R-packages/evalcast/vignettes/intro_vignette.Rmd

```diff
@@ -0,0 +1,132 @@
+---
+title: "Introduction to evalcast"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
+```
+
+The `evalcast` package provides the infrastructure for developing and evaluating probabilistic forecasters that are based on data obtained with the [covidcast](https://cmu-delphi.github.io/covidcast/covidcastR/) R package.  A unique feature of the `covidcast` API is that it can retrieve data that would have been available *as of* a certain date.  This accounts for a difficult property of working with certain COVID-19 data sources, which is that they may be backfilled (retrospectively updated).  Failing to account for backfill can lead to poorly trained forecasters and overly optimistic retrospective evaluations.  The `evalcast` package is designed to help forecasters avoid these pitfalls.
+
+
+## Testing forecasters
+We will use the included `baseline_forecaster` to demonstrate how to backtest a forecaster and compare its predictions to contemporaneous forecasts submitted to CovidHub. While `baseline_forecaster` is very simple, it can be used as a template for creating more sophisticated forecasters.
+
+The format of a forecaster is aligned with the [CovidHub submission instructions](https://github.com/reichlab/covid19-forecast-hub/blob/master/data-processed/README.md) and with the [covidcast](https://cmu-delphi.github.io/covidcast/covidcastR/) R package.
+
+### Specify the signals that will be used by forecaster
+
+We start by specifying which signals the forecaster will be using.  See [here](https://cmu-delphi.github.io/delphi-epidata/api/covidcast_signals.html) for a full list of signals available through the `covidcast` API.  Each signal is specified by two strings: `data_source` and `signal`.  Optionally, we can specify how far back in time we will want data.
+
+```{r}
+library(tidyverse)
+signals <- tibble(data_source = "jhu-csse", 
+                  signal = c("deaths_incidence_num", "confirmed_incidence_num"), 
+                  start_day = "2020-06-15")
+signals
+```
+
+### Run the forecaster on some dates
+
+Within the `evalcast` framework, forecasters do not directly retrieve historic data. Instead, `get_predictions` provides *only the data that would have been available at the time* to the specified forecaster in order to generate retrospective predictions.
+
+```{r}
+library(evalcast)
+library(covidcast)
+library(lubridate)
+
+forecast_dates <- as_date(get_forecast_dates("CMU-TimeSeries"))
+forecast_dates_july <- forecast_dates[forecast_dates >= "2020-07-01" & forecast_dates <= "2020-07-31"]
```

Comment:
I think this would be cleaner:
```
forecast_dates_july <- str_subset(forecast_dates, "^2020-07")
```
Also, we might consider going more recent at this point than July.  Perhaps September or October?